activestorage/lib/active_storage/engine.rb

```diff
@@ -34,8 +34,8 @@ class Engine < Rails::Engine # :nodoc:
     end
 
     initializer "active_storage.services" do
-      config.after_initialize do |app|
-        if config_choice = app.config.active_storage.service
+      config.to_prepare do
+        if config_choice = Rails.configuration.active_storage.service
           config_file = Pathname.new(Rails.root.join("config/storage.yml"))
```

Comment:
Good point. Maybe we could extract the file read behavior from config_for to a private API method and reuse here?